pkg/alertmanager/statefulset.go

```diff
@@ -675,6 +676,35 @@ func makeStatefulSetSpec(logger log.Logger, a *monitoringv1.Alertmanager, config
 		}
 	}
 
+	// If the am version >= 0.24.0, create a new secret for the mTLS configuration.
+	// The secret volume is always created and mounted, even if the secret is empty.
+	// But, empty volume cannot be passed as the path in the argument for the cluster TLS configuration.
+	// So, we only pass the argument `--cluster.tls-config` if the ClusterTLSConfig field is not nil.
+	if version.GTE(semver.MustParse("0.24.0")) {
+		var mtlsFields *monitoringv1.AlertmanagerTLSConfigFields
+		var isEmpty bool
+		if a.Spec.ClusterTLSConfig != nil {
+			isEmpty = false
+			mtlsFields = a.Spec.ClusterTLSConfig
+		}
```

Comment:
```suggestion
		clusterTLS := a.Spec.ClusterTLSConfig
```

Reply:
Woah! How did I find the most roundabout way to do this :)